packages/airnode-protocol/scripts/verify-local.ts

```diff
@@ -9,14 +9,44 @@ import { logger } from '@api3/airnode-utilities';
 import { contractNames } from './contract-names';
 const hre = require('hardhat');
 
+const verifyBytecode = (deployedData: string, localData: string, contractName: string) => {
+  if (deployedData === localData) {
+    logger.log(`✅  ${contractName} undeterministic deployment on ${hre.network.name} matches the local build!`);
+    return true;
+  }
+  if (deployedData === hre.ethers.constants.HashZero + localData.slice(2)) {
+    logger.log(`✅  ${contractName} deterministic deployment on ${hre.network.name} matches the local build!`);
+    return true;
+  }
+  // The compiler appends by default the IPFS hash of the metadata file to the end of the bytecode (the last 106 char)
+  // for reference: https://docs.soliditylang.org/en/v0.8.17/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode
+  if (deployedData.slice(0, -106) === localData.slice(0, -106)) {
+    logger.log(`✅  ${contractName} undeterministic deployment on ${hre.network.name} matches the local build!`);
+    logger.log(`⚠️  ${contractName} metadata is different from onchain value`);
+    return true;
+  }
+  if (deployedData.slice(0, -106) === hre.ethers.constants.HashZero + localData.slice(2).slice(0, -106)) {
+    logger.log(`✅  ${contractName} deterministic deployment on ${hre.network.name} matches the local build!`);
+    logger.log(`⚠️  ${contractName} metadata is different from onchain value`);
+    return true;
+  }
+  return false;
+};
+
 async function main() {
   for (const contractName of contractNames) {
     const deployment = await hre.deployments.get(contractName);
     const artifact = await hre.deployments.getArtifact(contractName);
 
     const creationTx = await hre.ethers.provider.getTransaction(deployment.transactionHash);
+    const deterministicDeploymentAddress = hre.ethers.utils.getCreate2Address(
+      '0x4e59b44847b379578588920ca78fbf26c0b4956c', // default create2 factory address in hardhat
+      hre.ethers.constants.HashZero,
+      hre.ethers.utils.keccak256(artifact.bytecode)
+    );
+
     // Verify that the creation tx hash belongs to the address
-    assert(creationTx.creates === deployment.address);
+    assert(creationTx.creates === deployment.address || deterministicDeploymentAddress);
```

Comment:
I think you intended `creationTx.creates === deployment.address || creationTx.creates === deterministicDeploymentAddress` here, yet this still accidentally creates the same outcome. `creationTx.creates` returns `null` when the deployment is deterministic, so `||` falls back to `deterministicDeploymentAddress`, which is treated as `true` because it's defined (though there is no case where it is not defined so this feels off).
I'll push a refactor.

Reply:
right, the assertion will always be true in this case. I guess the correct assertion would be ```creationTx.creates === deployment.address || deterministicDeploymentAddress === deployment.address```